### PR TITLE
Move Python 2-only stubs to @python2 directory

### PR DESCRIPTION
**Pull request by srittau**
Part of #5630

### CODE REVIEW

**Comment by srittau** [comment]
Could I excite someone to review this? The 47 files changed files may look intimidating, but apart from renames, it's just:

* documentation changes
* updates to the tests
* removing now redundant `python2`/`python3` fields
* using `Any` for non-existing imports, instead of `# type: ignore`ing the import (which is admittedly out of scope for this PR, but I found the workaround faster than I found the exclude list)

If you don't think this is a good idea for some reason, please let me know (in #5630 preferably).

File: tests/mypy_test.py [hunk 1]
```diff
@@ -101,13 +103,18 @@ def parse_version(v_str):
...
         # Python 2 is not supported by default.
-        return bool(data.get("python2", False))
+        return bool(data.get("python2", False)) or (dist_path / "@python2").exists()
     # Python 3 is supported by default.
-    return bool(data.get("python3", True))
+    return has_py3_stubs(dist_path)
+
+
+def has_py3_stubs(dist: Path) -> bool:
+    return len(glob(f"{dist}/*.pyi")) > 0 or len(glob(f"{dist}/[!@]*/__init__.pyi")) > 0
```
Comment (Akuli) [inline tests/mypy_test.py:117]: On the one hand, I don't like how this is copy/pasted from `stubtest_third_party.py`, but on the other hand, it's not really worth putting this into a separate file and import it. Maybe add a comment to both places mentioning the function is copied?

File: tests/mypy_test.py [hunk 2]
```diff
@@ -18,6 +18,8 @@
 import sys
 import toml
 import tempfile
+from glob import glob
+from pathlib import Path
 from typing import Dict, NamedTuple
 
 PY2_NAMESPACE = "@python2"
```
Comment (Akuli) [inline tests/mypy_test.py:25]: This doesn't seem to be worth keeping in a global constant. It's already hard-coded in a couple places.

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -59,8 +59,10 @@ go into `stubs`. Each subdirectory there represents a PyPI distribution, and
 contains the following:
 * `METADATA.toml`, describing the package. See below for details.
 * Stubs (i.e. `*.pyi` files) for packages and modules that are shipped in the
-  source distribution. If the Python 2 version of the stubs must be kept
-  *separate*, they can be put in a `@python2` subdirectory.
+  source distribution.
+* If the stubs are either Python 2-only, or if the Python 2 and Python 3 stubs
+  are separate, the Python 2 stubs are put in a `@python2` subdirectory. In the
+  former case, there are no stubs or package directories on the top level.
```
Comment (Akuli) [inline CONTRIBUTING.md:65]: If I understood this correctly, we could just say this instead, making it more obvious where to find the Python 3 stubs:

```suggestion
  are separate, the Python 2 stubs are put in a `@python2` subdirectory.
  Stubs or package directories at the top level are always used with Python 3,
  so there are no stubs or package directories on the top level of Python 2-only stubs.
```

**Comment by JelleZijlstra** [comment]
Sorry, I merged before I saw @Akuli's comments.

**Comment by Akuli** [comment]
I'll make a new PR of what I suggested.

File: tests/mypy_test.py (hunk 1 above)
Comment (srittau) [reply tests/mypy_test.py:117]: We have similar code snippets in mypy and pytype. Maybe it's worth thinking about a small `typeshed` module.

**Pull request merged**